90.md

```diff
@@ -0,0 +1,438 @@
+    ]
+}
+```
+
+* `request`: The job request event stringified-JSON.
+* `amount`: millisats that the Service Provider is requesting to be paid. An optional third value can be a bolt11 invoice.
+
+## Job feedback
+Service providers can give feedback about a job back to the customer.
+
+```json
+{
+    "kind": 65000,
+    "content": "<empty-or-payload>",
+    "tags": [
+        [ "status", "<status>", "<extra-info>" ],
+        [ "amount", "requested-payment-amount", "<bolt11>" ],
+        [ "e", "<job-request-id>", "<relay-hint>" ],
+        [ "p", "<customer's-pubkey>" ],
+    ]
+}
+```
+
+* `content`: Either empty or a job-result (e.g. for partial-result samples)
+* `status` tag: Service Providers SHOULD indicate what this feedback status refers to. [Appendix 3](#appendix-3-job-feedback-status) defines status. Extra human-readable information can be added as an extra argument.
+* `amount` tag: as defined in the [Job Result](#job-result) section.
+
+# Protocol Flow
+* Customer publishes a job request (e.g. `kind:65002` speech-to-text).
+* Service Providers can submit `kind:65000` job-feedback events (e.g. `payment-required`, `processing`, `error`, etc.).
+* Upon completion, the service provider publishes the result of the job with a `kind:65001` job-result event.
+* At any point, if there is an `amount` pending to be paid as instructed by the service provider, the user can pay the included `bolt11` or zap the job result event the service provider has sent to the user
+
+`kind:65000` and `kind:65001` events MAY include an `amount` tag, this can be interpreted as a suggestion to pay. Service Providers SHOULD use the `payment-required` feedback event to signal that a payment is required and no further actions will be performed until the payment is sent. Customers can always either pay the included `bolt11` invoice or zap the event requesting the payment and service providers should monitor for both if they choose to include a bolt11 invoice.
+
+## Notes about the protocol flow
+The flow is deliberately ambiguous, allowing vast flexibility for the interaction between customers and service providers so that service providers can model their behavior based on their own decisions/perceptions of risk.
+
+Some service providers might choose to submit a `payment-required` as the first reaction before sending a `processing` or before delivering `kind:65001` results, some might choose to serve partial results for the job (e.g. as a sample), send a `payment-required` to deliver the rest of the results, and some service providers might choose to assess likelihood of payment based on an npub's past behavior and thus serve the job results before requesting payment for the best possible UX.
+
+It's not up to this NIP to define how individual vending machines should choose to run their business.
+
+# Cancellation
+A job request might be cancelled by publishing a `kind:5` delete request event tagging the job request event.
```

Comment:
Depending on relay implementation, this might never get received, and the service provider might only have a missing event to go off of. Maybe #669 would help.